beanquery/query_render.py

```diff
@@ -257,39 +258,51 @@ def format(self, number, key=None):
 
 
 class AmountRenderer(ColumnRenderer):
-    """A renderer for amounts. The currencies align with each other.
+    """Renderer for Amount instances.
+
+    The numerical part is formatted with the right quantization
+    determined by ``dcontext`` in the rendering context and aligned on
+    the decimal point across rows. Numbers are right padded with
+    spaces to alignt the commodity symbols across rows::
+
+      -  1234.00   USD
+      -    42      TEST
+      -     0.0001 ETH
+      -   567.00   USD
+
     """
     dtype = amount.Amount
 
     def __init__(self, ctx):
-        self.rdr = DecimalRenderer(ctx)
-        self.ccylen = 0
+        super().__init__(ctx)
+        # Use the display context inferred from the input ledger to
+        # determine the quantization of the column values.
+        self.quantize = ctx.dcontext.quantize
+        # Use column specific display context for formatting.
+        self.dcontext = display_context.DisplayContext()
```

Comment:
@blais After some thinking this is the think I came up with to format Amounts in query results: it uses the display context inferred from the input ledger to quantize the numbers but uses a column specific DisplayContext instance to do the formatting. In this way the query output does not influence the display of subsequent queries. The tests think that it works.